### DISCUSSION TITLE
Install on RHEL7.8 had trouble finding git and curl in /usr/local; kluged around

### DISCUSSION THREAD

**Discussion by keshlam** [Homebrew/discussions#254] [Linux]
I have the RHEL git and curl installed in /usr/bin, but Homebrew insisted it needed newer versions when it got to the self-updating step.

So I manually built/installed newer versions of both those tools, installed them in /usr/local/bin, and confirmed that they were now on my path and being resolved by _where_ correctly.

Homebrew's update still insisted on complaining about the old versions being present; for some reason it wasn't finding the new ones even though they should have been earlier in the path.

I was able to kluge past this by setting
```
   export HOMEBREW_GIT_PATH=/usr/local/bin/git
   export HOMEBREW_CURL_PATH=/usr/local/bin/curl
   export HOMEBREW_DEVELOPER=cheat

```
before attempting the install... but I really shouldn't have had to manually override those.

I haven't yet attempted to debug why the wrong version is being found by default; right now my priority is to finish the installation I needed brew for in the first place. If I come up for air some time in the next month or two I can try to track this down for you... but you probably shouldn't wait for me.

**Comment by maxim-belkin** [comment]
Hi Joe,

Welcome to Homebrew and thanks for the report.

What you did is exactly right and this is what one has to do in order to use Homebrew on a Linux system with (what Homebrew considers) "outdated" cURL and Git. I do exactly the same on one of my Linux systems. This is the consequence of using "clean" environment -- you can check it out with `brew sh` (interactive mode) or `brew sh -c 'env'`. 

Unfortunately, that's the down side of using a macOS-centric solution with its own definition of "outdated". But I'm sure we'll be able to change this as more and more of us (Linux people) use Homebrew and submit reports like this one. In fact, I proposed to notify users of a way to resolve this -- see, for example [this PR](https://github.com/Homebrew/brew/pull/9213/files/22bc5a94e71ba76d0ffb5ce3d959912e42d8dce5) -- and I think your report counts as a request that @MikeMcQuaid mentioned [here](https://github.com/Homebrew/brew/pull/9213/files/22bc5a94e71ba76d0ffb5ce3d959912e42d8dce5#r528529983). I'm going to mention this discussion there so that we can keep track of reports like this one.

Again, thanks for the note! 👍

Maxim

**Reply by MikeMcQuaid** [reply]
To me the best solution here would be figuring out how to get Homebrew's `curl` and `git` to self-bootstrap with really old versions.

**Comment by keshlam** [comment]
Good to know it's not just me...Calling this out a bit more strongly in the Homebrew installation hints might have saved my having to dig into the scripts. I hate saying "fix it in pubs", but that approach does have near-zero risk of new breakage.Homebrew, once over this hump, did do what I needed it to. My thanks to everyone who has gotten it this far!______________________________________"Your data is important to us. Please stay on the port, and your connection will be accepted by the next available thread."

**Comment by atwok** [comment]
Adding my vote for hinting to the user that there is way to point "brew" to an updated version of curl.  I went so far as to use miniconda3 just to install curl and git, but even after doing so and putting them in my path, I still received the same errors about curl and git from brew (and now I know why! :))

Also, are the various command line overrides that one can use with "brew" documented anywhere?  It wasn't until I used the magic google searching combo that I ran across this thread and found out about GIT_PATH, CURL_PATH, and DEVELOPER.

**Reply by MikeMcQuaid** [reply]
https://github.com/Homebrew/brew/pull/10796 is ongoing to add this as documented behaviour
